engine/src/player.ts

```diff
@@ -296,13 +297,13 @@ export default class Player extends EventEmitter {
     return this.data.occupied.filter((hex) => hex.data.planet !== Planet.Empty && hex.isMainOccupier(this.player));
   }
 
-  loadFaction(faction: Faction, expansions = 0, skipIncome = false) {
+  loadFaction(faction: Faction, customization: FactionCustomization, expansions = 0, skipIncome = false) {
     this.faction = faction;
-    this.board = factionBoard(faction);
+    this.board = factionBoard(faction, customization);
```

Comment:
```suggestion
    this.boardCustomization = this.boardCustomization || factionBoardCustomization(faction, customization)
    this.board = factionBoard(faction, this.boardCustomization);
```

So, the precise customization is stored in the JSON of the player. If it changes in the future and an old game is replayed, there will be no problem.

The downside is it can only store JSON, not javascript functions, but the customizations for now don't include javascript functions.

Reply:
what should the type of `boardCustomization` be? `FactionBoardRaw`?

Reply:
`Partial<FactionBoardRaw>` or `PartialDeep<FactionBoardRaw>`

The latter can be obtained with:

```
import type { PartialDeep } from "type-fest"
```

Reply:
why not `FactionBoardRaw` - all the fields are optional there..

Reply:
you're right!